tests/sles4sap/microsoft_sdaf/sdaf_deploy_workload_zone.pm

```diff
@@ -0,0 +1,46 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+# Maintainer: QE-SAP <qe-sap@suse.de>
+# Summary: Deployment of the workload zone and SUT using SDAF automation
+
+use parent 'sles4sap::microsoft_sdaf_basetest';
+
+use strict;
+use warnings;
+use sles4sap::sdaf_library;
+use sles4sap::console_redirection;
+use serial_terminal qw(select_serial_terminal);
+use testapi;
+
+sub test_flags {
+    return {fatal => 1};
+}
+
+sub run {
+    serial_console_diag_banner('workload zone deployment: start');
+
+    # From now on everything is executed on Deployer VM (residing on cloud).
+    connect_target_to_serial();
+    load_os_env_variables();
+
+    # Setup Workload zone openQA variables - used for tfvars template
+    set_var('SDAF_RESOURCE_GROUP', generate_resource_group_name(deployment_type => 'workload_zone'));
+    set_var('SDAF_VNET_CODE', get_required_var('SDAF_WORKLOAD_VNET_CODE'));
+    # 'vnet_code' variable changes with deployment type.
+    set_os_variable('vnet_code', get_required_var('SDAF_WORKLOAD_VNET_CODE'));
+    prepare_tfvars_file(deployment_type => 'workload_zone');
+    az_login();
+    sdaf_deploy_workload_zone();
+
+    # diconnect the console
+    disconnect_target_from_serial();
+
+    # reset temporary variables
+    set_var('SDAF_RESOURCE_GROUP', '');
```

Comment:
is it ok or should be 
```suggestion
    set_var('SDAF_RESOURCE_GROUP', undef);
```

Reply:
Yea I was thinking about this as well. I am just not sure how does openqa handle 'undef' value. You know, if by chance it takes it as a string. I'll check once I am back home.

Reply:
`testapi::set_var()` does:
```
sub set_var ($var, $val, %args) {
    $bmwqemu::vars{$var} = $val;
```
So if `$val` is `undef`, it should work just as well. However, both `''` and `undef` will evaluate to false when you do `get_var('SDAF_RESOURCE_GROUP')` so I think in the end it's the same.